### PR TITLE
fix: add validation for CRP and PickFixed cluster names

### PR DESCRIPTION
**Pull request by nwnt**
### Description of your changes

As a result from the latest bug bash, we want to introduce validations for:
  - the CRP resource name, which currently only validates the length but not the characters in the name.
  - the cluster names for PickFixed policy.

I have:

- [X] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested
 
Run unit tests and add new test cases for negative validation scenarios.

### CODE REVIEW

File: pkg/utils/validator/clusterresourceplacement.go [hunk 1]
```diff
@@ -101,8 +101,9 @@ func ValidateClusterResourcePlacementAlpha(clusterResourcePlacement *fleetv1alph
 func ValidateClusterResourcePlacement(clusterResourcePlacement *placementv1beta1.ClusterResourcePlacement) error {
 	allErr := make([]error, 0)
 
-	if len(clusterResourcePlacement.Name) > validation.DNS1035LabelMaxLength {
-		allErr = append(allErr, fmt.Errorf("the name field cannot have length exceeding %d", validation.DNS1035LabelMaxLength))
+	crpNameErr := validation.IsDNS1035Label(clusterResourcePlacement.Name)
```
Comment (ryanzhang-oss) [inline pkg/utils/validator/clusterresourceplacement.go]: this is not correct, we are not enforcing IsDNS1035Label for the CRP name. We are only enforcing the length of the name.

Comment (nwnt) [reply pkg/utils/validator/clusterresourceplacement.go]: removed this part.

File: pkg/utils/validator/clusterresourceplacement.go [hunk 2]
```diff
@@ -194,6 +194,10 @@ func validatePolicyForPickFixedPlacementType(policy *placementv1beta1.PlacementP
 	}
 	uniqueClusterNames := make(map[string]bool)
 	for _, name := range policy.ClusterNames {
+		nameErr := validation.IsDNS1123Label(name)
```
Comment (ryanzhang-oss) [inline pkg/utils/validator/clusterresourceplacement.go]: actually it should be IsDNS1123Subdomain and length(name) < 64

Comment (nwnt) [reply pkg/utils/validator/clusterresourceplacement.go]: changed to what you suggested now.
